internal/mcpsrv/file_source.go

```
@@ -0,0 +1,206 @@
+		return "", fileSource{}, fmt.Errorf("resolve %q: %w", path, err)
+	}
+	if !withinRoots(resolved, roots) {
+		return "", fileSource{}, fmt.Errorf(
+			"%q is outside ANTI_TANGENT_PLAN_ROOTS (%s)", resolved, strings.Join(roots, string(os.PathListSeparator)))
+	}
+
+	f, err := openNoFollow(resolved)
+	if err != nil {
+		return "", fileSource{}, fmt.Errorf("open %q: %w", resolved, err)
+	}
+	defer f.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `f.Close` is not checked

(errcheck)